File: objectiveai-sdk-py/objectiveai_sdk/vector/completions/http.py
"""HTTP functions for vector completion endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING, Union

if TYPE_CHECKING:
    from objectiveai_sdk.client import ObjectiveAI
    from objectiveai_sdk.stream import Stream
    from objectiveai_sdk.vector.completions.request import VectorCompletionCreateParams
    from objectiveai_sdk.vector.completions.response.streaming import VectorCompletionChunk
    from objectiveai_sdk.vector.completions.response.unary import VectorCompletion


async def create_vector_completion(
    client: ObjectiveAI,
    params: VectorCompletionCreateParams,
) -> Union[VectorCompletion, Stream[VectorCompletionChunk]]:
    """Create a vector completion.

    If ``params.stream`` is true, returns a streaming response.
    Otherwise returns the complete response.
    """
    if getattr(params, "stream", None):
        return await client.post_streaming("vector/completions", params)
    return await client.post_unary("vector/completions", params)

File: objectiveai-sdk-rs/src/vector/completions/http.rs
//! HTTP client functions for vector completions.

use crate::{HttpClient, HttpError, McpHandler, Notifier};
use futures::Stream;

/// Creates a vector completion and waits for the complete response.
///
/// Sets `stream: None` to ensure a unary response.
pub async fn create_vector_completion_unary(
    client: &HttpClient,
    mut params: super::request::VectorCompletionCreateParams,
) -> Result<super::response::unary::VectorCompletion, HttpError> {
    params.stream = None;
    client
        .send_unary(reqwest::Method::POST, "vector/completions", Some(params))
        .await
}

/// Creates a vector completion with streaming response.
///
/// Opens a WebSocket and returns `(Stream<Chunk>, Notifier)`. See
/// [`crate::agent::completions::http::create_agent_completion_streaming`]
/// for the full semantics of `handler` and the demux.
pub async fn create_vector_completion_streaming<H: McpHandler>(
    client: &HttpClient,
    mut params: super::request::VectorCompletionCreateParams,
    handler: H,
) -> Result<
    (
        impl Stream<
            Item = Result<
                super::response::streaming::VectorCompletionChunk,
                HttpError,
            >,
        > + Send
        + Unpin
        + 'static
        + use<H>,
        Notifier,
    ),
    HttpError,
> {
    params.stream = Some(true);
    client
        .send_streaming_ws(
            reqwest::Method::POST,
            "vector/completions",
            params,
            handler,
        )
        .await
}
